extensions/amp-form/0.1/amp-form.js

```diff
@@ -88,11 +92,14 @@ export class AmpForm {
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
-          .catch(error => {
-            this.setState_(FormState_.SUBMIT_ERROR);
-            rethrowAsync('Form submission failed:', error);
-          });
+      }).then(response => {
+        this.setState_(FormState_.SUBMIT_SUCCESS);
+        this.renderTemplate_(FormState_.SUBMIT_SUCCESS, response);
+      }).catch(error => {
+        this.setState_(FormState_.SUBMIT_ERROR);
+        this.renderTemplate_(FormState_.SUBMIT_ERROR, error).then(
+            rethrowAsync.bind(null, 'Form submission failed:', error));
```

Comment:
I found this work better for testing. I don't really understand exactly why but basically If I rethrowAsync sync after the `this.renderTemplate` call, the test always fail with the error being thrown, if I try to expect that test to eventually throw it would catch the error but it doesn't seem to test the last block of the test which does the actual expectations inside the `timer.promise(0).then`.

I can try and debug this some more if you think we should do this sync.